src/modules/services_manager_service.rb

```diff
@@ -20,9 +20,13 @@ class ServicesManagerServiceClass < Module
       manual:    N_('Manually')
     }.freeze
 
+    # [Boolean] Set whether the module has been modified
+    attr_writer :modified
+
     # @return [Hash{String => Yast2::SystemService}]
     def services
-      @services ||= read
+      read if @services.nil?
```

Comment:
NP: Why not simply `read` ?

Reply:
Anyway, why to move the memorization step to the `read` method? I would expect that `read` always perform the action.

Reply:
Ah, I see, we have `#read` and `#reload`. So `#read` is basically an alias of `#services`. BTW, we are changing the behavior of `#read`. Could it be a problem?

Reply:
No, I do not expect it to be a problem. Now it behaves more like other `#read` methods in YaST/AutoYaST.